contributor-docs/how_firebase_works.md

```diff
@@ -0,0 +1,81 @@
+# How Firebase Works
+
+## Background
+
+### Eager Initialization
+
+One of the biggest strengths for Firebase clients is the ease of integration. In a common case, a developer has very few things to do to integrate with Firebase. There is no need to initialize/configure Firebase at runtime. Firebase automatically initializes at application start and begins providing value to developers. A few notable examples:
+
+* `Analytics` automatically tracks app events
+* `Firebase Performance` automatically tracks app startup time, all network requests and screen performance
+* `Crashlytics` automatically captures all application crashes, ANRs and non-fatals
+
+The ability to initialize and start operating automatically is a core feature of Firebase that makes onboarding and adoption very simple. However with this advantage comes a great responsibility to keep the application snappy and not to slow down application startup for 3p developers as it can stand in the way of user adoption for the application.
```

Comment:
```suggestion
This feature makes onboarding and adoption very simple. However, comes with the great responsibility of keeping the application snappy. We shouldn't slow down application startup for 3p developers as it can stand in the way of user adoption of their application.
```